src/wagtail_factories/factories.py

```diff
@@ -146,3 +147,31 @@ class Meta:
 
     title = "A document"
     file = factory.django.FileField()
+
+
+class RedirectFactory(DjangoModelFactory):
+    old_path = factory.Faker("slug")
+    site = factory.SubFactory(SiteFactory)
+    redirect_page = factory.SubFactory(PageFactory)
+    redirect_link = factory.Faker("url")
+    is_permanent = factory.Faker("boolean")
+
+    class Meta:
+        model = Redirect
+
+    @classmethod
+    def _create(cls, model_class, *args, **kwargs):
+        """
+        Override _create() to ensure that Redirect.clean() is run
+        in order to normalise `old_path`.
+        """
+        obj = model_class(*args, **kwargs)
+        try:
+            obj.clean()
+        except:
```

Comment:
Issue: what errors are we catching here? A validation error, for example, should be allowed to propagate, and raising a `RuntimeError` would only obfuscate the details.

Reply:
https://github.com/wagtail/wagtail-factories/issues/82 has the updated version. We're catching a `ValidationError` here

Reply:
I'm not sure if we need the `ValidationError` checks. They were there because the original code was using `full_clean()` rather than `clean()`. Since that `clean()` doesn't actually do any validation, I don't think there's potential for validation errors here? https://github.com/wagtail/wagtail/blob/633004e61b59436509e7b6fbe2a7c9a8eb440e46/wagtail/contrib/redirects/models.py#L191C12-L191C12